10up/block-catalog
# Add workflow to run PHPUnit tests

### Description of the Change

Adds a workflow to run PHPUnit tests on any PR that has changed PHP files.

Note that though the issue this is linked to also mentions adding in an E2E workflow, we currently don't have any E2E tests so I think that should be handled separately.

This Action does the following:

- Runs anytime a PR is opened against develop or code is pushed to develop or trunk
- Only runs if PHP files have been changed
- Sets up PHP version 7.4 (our minimum), runs composer install, installs the WordPress test suite and then runs our PHPUnit tests

Closes #14

### How to test the Change

Can see a commit with no PHP changes here: https://github.com/10up/block-catalog/pull/24/checks?sha=3288c66c43ade07171f602bb740db23c6fff9a44

Can see a commit that breaks one of our functions and the unit test fails: https://github.com/10up/block-catalog/pull/24/checks?sha=76522f33c2c1351bc3789f9228831be5b6dc4965

### Changelog Entry

> Added - GitHub Action workflow to run PHPUnit tests on all needed PRs.

### Credits

Props @dkotter

### Checklist:
